packages/server/src/auth/profile.test.ts

```diff
@@ -34,12 +34,19 @@ describe('Profile', () => {
         password,
       });
 
+      const otherProfile = await systemRepo.createResource<ProfileResource>({
+        resourceType: 'Practitioner',
+        meta: { project: registerResult.project.id },
+        name: [{ text: 'Other' }],
+      });
+
       const inviteResult = await inviteUser({
         project: registerResult.project,
         resourceType: 'Practitioner',
         firstName: 'Multi2',
         lastName: 'Multi2',
         email,
+        membership: { profile: createReference(otherProfile) },
```

Comment:
I'm not sure if this will work with the login flow. 

[this line ](https://github.com/medplum/medplum/blob/66f331240d7dc0aa776bd7eef810695a93f8e9cb/packages/server/src/oauth/utils.ts#L179-L188) only handles the case where there are 0 or 1 memberships per (user, project).  

With two memberships, it returns a `Login` with 0 memberships, which ends up generating a 500 down the line

Reply:
It works the same way as any `User` with multiple `ProjectMembership` resources - it returns the partial `Login` + list of profiles for the user to choose from.

Reply:
That will work when using our `SignInForm` component (or calling `/auth/login`) But it breaks the SSO flow through the token exchange, which is the flow the user was using for the original ticket. 

https://github.com/medplum/medplum/blob/66f331240d7dc0aa776bd7eef810695a93f8e9cb/packages/server/src/oauth/token.ts#L387-L390